ocaml/xapi/helpers.ml

```diff
@@ -1021,15 +1021,20 @@ let is_valid_MAC mac =
          acc && String.length s = 2 && validchar s.[0] && validchar s.[1])
        true l
 
-(** Returns true if the supplied IP address looks like one of mine *)
+(** Returns true if the supplied address looks like one of mine *)
 let this_is_my_address ~__context address =
   let dbg = Context.string_of_task __context in
   let inet_addrs =
     Net.Interface.get_ipv4_addr dbg
       (Xapi_inventory.lookup Xapi_inventory._management_interface)
   in
   let addresses = List.map Unix.string_of_inet_addr (List.map fst inet_addrs) in
-  List.mem address addresses
+  let other_names =
+    let hostname = Unix.gethostname () in
+    let fqdns = Gencertlib.Lib.fqdns_of_hostname hostname in
+    hostname :: fqdns
+  in
+  List.mem address (List.append addresses other_names)
```

Comment:
I think changing the names of the list to `ipv4s` and `hostnames` would help with readability (ipv6 addresses are missing here, pinging @benjamreis)

Reply:
Should I add it in my PR? after a rebase.

Reply:
@benjamreis You won't want to rebase on top of this branch because it includes feature work, but indeed it would be good to make this function work correctly in the ipv6 case

Reply:
Sure, I can make a new PR pointing to this one. I think it's the simpler solution. What do you think?

Reply:
Hmm, for some reason I can't make a PR towards `lippirk/pool-conf`. How can I supply my patch?

Reply:
It will take months until this work is merged to master. We merge master to the feature branch from time you time. I'd recommend to have the patch done against master and we'll resolve the conflict when merging master to the feature branch, it should be an easy conflict to resolve.

Reply:
https://github.com/xapi-project/xen-api/pull/4261 here you go.